ext/javy/javy.rb

```diff
@@ -0,0 +1,170 @@
+require "rbconfig"
+require "open-uri"
+require "zlib"
+require "open3"
+
+class Javy
+  ROOT = __dir__
+  TARGET = File.join(ROOT, "javy")
+  VERSION = File.read(File.join(ROOT, "version")).strip
+
+  def self.install(**args)
+    new.install(**args)
+  end
+
+  def self.build(**args)
+    new.build(**args)
+  end
+
+  def install(**args)
+    Install.call(target: target, platform: platform, version: VERSION, **args)
+  end
+
+  def build(source:, dest:)
+    exec(source, "-o", dest)
+  end
+
+  private
+
+  def exec(*args, **kwargs)
+    system(target, *args, **kwargs)
```

Comment:
Have you considered using one of the `capture*` methods from [open3](https://docs.ruby-lang.org/en/2.0.0/Open3.html#method-i-capture2e)? I noticed some outputs from Javy like the following and wonder if we want to hide that from the user:
```
warning: unknown name subsection with id 9 at 2875902
```

Reply:
I was originally hoping to make this a very light wrapper to avoid adding in any ruby/cli magic. However, it ended up being more convenient to have access to stderr later on in another PR, so I ended up using `capture2e` which captures the output and returns stdout/stderr!